### PR TITLE
Add token metrics

### PR DESCRIPTION
**Pull request by italolelis**
## What does this PR do?

Adds more metrics to token validation. If it's a JWT we can be very precise about which kind of error to track on the metrics. This will allow you to build dashboards with metrics categories for each type of error.

### CODE REVIEW

File: pkg/metrics/metrics_context.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// NewContext returns a context that has a stats Client
+func NewContext(ctx context.Context, client stats.Client) context.Context {
+	return context.WithValue(ctx, statsKey, client)
+}
+
+// WithContext returns a stats Client with as much context as possible
+func WithContext(ctx context.Context) stats.Client {
+	ctxStats, ok := ctx.Value(statsKey).(stats.Client)
+	if !ok {
+		ctxStats, _ := stats.NewClient("noop://", "")
```
Comment (vgarvardt) [inline pkg/metrics/metrics_context.go:22]: Can you bump `stats-go` to the latest version here or should we do this in separate PR?

Comment (italolelis) [reply pkg/metrics/metrics_context.go:22]: Lets do it in a separate PR

**Review by vgarvardt — APPROVED** [review]
